rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -583,7 +580,7 @@ static void discovery_thread(rmw_context_impl_t * impl)
   dds_entity_t ws;
   /* deleting ppant will delete waitset as well, so there is no real need to delete
      the waitset here on error, but it is more hygienic */
-  if ((ws = dds_create_waitset(impl->ppant)) < 0) {
+  if ((ws = dds_create_waitset(DDS_CYCLONEDDS_HANDLE)) < 0) {
```

Comment:
That phrase applies to lot of places in `rmw_cyclonedds`, `rmw_fastrtps` and `rmw_connext`, so I will ignore it for the moment.

Reply:
Also, in Cyclone you’re dealing with integers so I don’t know how easy it would be to retrofit that (I know you can wrap pointers in `unique_ptr`, perhaps there are other tricks as well). Rewriting it to use the “new” OMG C++ DDS API would allow you to use RAII everywhere, but I’m not sure it’d be worth the bother to rewrite it.